tests/TestCase/Datasource/ConnectionTest.php

```diff
@@ -88,4 +89,23 @@ public function testQueryLogging()
 
         $this->assertNotEmpty($result);
     }
+
+    /**
+     * Ensure that logging queries works.
+     *
+     * @return void
+     */
+    public function testQueryLogger()
+    {
+        $logger = $this->getMockBuilder('Cake\Database\Log\QueryLogger')->setMethods(['log'])->getMock();
+        $logger->expects($this->once())->method('log');
+
+        $connection = ConnectionManager::get('test');
+        $connection->setLogger($logger);
+        $connection->logQueries(true);
+        $result = $connection->request('_stats');
+        $connection->logQueries(false);
+
+        $this->assertNotEmpty($result);
```

Comment:
Should there be assertions on the log message?

Reply:
Done!